File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int m=nums.size();
        int dp[m][2];
        dp[0][0]=0;
        dp[0][1]=nums[0];
        dp[1][0]=nums[0];
        dp[1][1]=max(nums[0],nums[1]);
        for(int i=2;i<m;i++)
        {
            dp[i][0]=dp[i-1][1];
            dp[i][1]=max((dp[i-1][0]+nums[i]),dp[i-2][0]+nums[i]);
        }
        return max(dp[m-1][0],dp[m-1][1]);

    }
};

File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n==1:
            return nums[0]
        a=[0,nums[0]]
        for i in range(1,n-1):
            a.append(max(a[-1],a[-2]+nums[i]))
        max1 = a[-1]
        a=[0,nums[1]]
        for i in range(2,n):
            a.append(max(a[-1],a[-2]+nums[i]))
        max2 = a[-1]
        return max(max1,max2)




        